pandas/tests/frame/test_analytics.py

```diff
@@ -693,82 +696,113 @@ def test_describe_tz_values(self, tz_naive_fixture):
+    # Reductions
 
-    def test_count(self, float_frame_with_na, float_frame, float_string_frame):
-        f = lambda s: notna(s).sum()
-        assert_stat_op_calc('count', f, float_frame_with_na, has_skipna=False,
-                            check_dtype=False, check_dates=True)
+    def test_stat_op_api(self, float_frame, float_string_frame):
         assert_stat_op_api('count', float_frame, float_string_frame,
                            has_numeric_only=True)
+        assert_stat_op_api('sum', float_frame, float_string_frame,
+                           has_numeric_only=True)
 
-        # corner case
-        frame = DataFrame()
-        ct1 = frame.count(1)
-        assert isinstance(ct1, Series)
+        assert_stat_op_api('nunique', float_frame, float_string_frame)
+        assert_stat_op_api('mean', float_frame, float_string_frame)
+        assert_stat_op_api('product', float_frame, float_string_frame)
+        assert_stat_op_api('median', float_frame, float_string_frame)
+        assert_stat_op_api('min', float_frame, float_string_frame)
+        assert_stat_op_api('max', float_frame, float_string_frame)
+        assert_stat_op_api('mad', float_frame, float_string_frame)
+        assert_stat_op_api('var', float_frame, float_string_frame)
+        assert_stat_op_api('std', float_frame, float_string_frame)
+        assert_stat_op_api('sem', float_frame, float_string_frame)
+        assert_stat_op_api('median', float_frame, float_string_frame)
 
-        ct2 = frame.count(0)
-        assert isinstance(ct2, Series)
+        try:
+            from scipy.stats import skew, kurtosis  # noqa:F401
+            assert_stat_op_api('skew', float_frame, float_string_frame)
+            assert_stat_op_api('kurt', float_frame, float_string_frame)
+        except ImportError:
+            pass
 
-        # GH 423
-        df = DataFrame(index=lrange(10))
-        result = df.count(1)
-        expected = Series(0, index=df.index)
-        tm.assert_series_equal(result, expected)
+    def test_stat_op_calc(self, float_frame_with_na, mixed_float_frame):
 
-        df = DataFrame(columns=lrange(10))
-        result = df.count(0)
-        expected = Series(0, index=df.columns)
-        tm.assert_series_equal(result, expected)
+        def count(s):
+            return notna(s).sum()
 
-        df = DataFrame()
-        result = df.count()
-        expected = Series(0, index=[])
-        tm.assert_series_equal(result, expected)
+        def nunique(s):
+            return len(algorithms.unique1d(s.dropna()))
 
-    def test_nunique(self, float_frame_with_na, float_frame,
-                     float_string_frame):
-        f = lambda s: len(algorithms.unique1d(s.dropna()))
-        assert_stat_op_calc('nunique', f, float_frame_with_na,
+        def mad(x):
+            return np.abs(x - x.mean()).mean()
+
+        def var(x):
+            return np.var(x, ddof=1)
+
+        def std(x):
+            return np.std(x, ddof=1)
+
+        def sem(x):
```

Comment:
what are all these for?

Reply:
These get passed to assert_stat_op_calc.  In the status quo these are defined in separate test methods and are mostly named `alt`